### DISCUSSION TITLE
Integration testing serve discovery not working

### DISCUSSION THREAD

**Discussion by bit365** [microsoft/aspire#4371] [Q&A]
host code:

```service1.WithReference(service2)```

service2 code：

```httpClient.GetAsync("https://service1")```

Running the HOST project directly is able to serve discovery and work properly, and service1 is resolved to the endpoint.

If you use DistributedApplicationTestingBuilder to start a test environment and perform integration tests on service2, httpClient.GetAsync("https://service1") fails to work, prompting an unknown service1 host and invalidating service discovery.

**Comment by ReubenBond** [comment]
@bit365 You have your reference reversed: if you want to call (reference) service1 from service2, service2 needs to reference service1: `service2.WithReference(service1)`.

Additionally, regarding testing, could you show how you're creating your HttpClient? It's different in testing for now: you need to call `app.CreateHttpClient("service1")` to get a client which is then configured with the base URI of "service1", so your requests should be relative to the base.

Here is some example code:

```csharp
var httpClient = app.CreateHttpClient("mywebapp1", "https");
var result = await httpClient.GetFromJsonAsync<WeatherForecast[]>("/weatherforecast");
```
